[Quek Zong Jin] iP

0wl3xSec opened the pull request:
# Pooh
> “People say nothing is impossible, but I do nothing every day.” —Winnie the Pooh 
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/77948749/6a21b018-f875-4065-9759-0ae861e86ea9)


**Pooh** frees your mind of having to remember things you need to do. It's,
- Text-based
- Easy to learn
- ~FAST~ **_SUPER SUPER_ FAST** to use

All you need to do is,

1. Download it from [here](https://github.com/Zjinnnn/ip/releases/tag/A-Jar). 🍯
2. Double-click it.
3. Add your tasks. 
4. Let it manage your tasks for you!

**What's the best about it?**
It is **FREE!**

**Features:**
- [X]  Managing tasks
- [X]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

---
For `Java` programmers, here's the `main` method:
```Java
public static void main(String[] args) {
        new Pooh("pooh.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

0wl3xSec committed:
feat: implement level-0 increment

0wl3xSec committed:
feat: implement level-1 increment

0wl3xSec committed:
feat: implement level-2 increment

0wl3xSec committed:
refactor: added Task class

0wl3xSec committed:
implement level-3 increment

0wl3xSec committed:
refactor: removed unnecessary code

0wl3xSec committed:
refactor: added Todo, Event, Deadline classes

0wl3xSec committed:
fix: missing }

0wl3xSec committed:
refactor: removed unnecessary parameters

0wl3xSec committed:
feat: implement level-4 increment

0wl3xSec committed:
style: removed extra white spaces

0wl3xSec committed:
test: Automated Text UI Testing

0wl3xSec committed:
refactor: added PoohException class

0wl3xSec committed:
feat: implement level-5 increment

0wl3xSec committed:
style: remove extra line

0wl3xSec committed:
feat: implement level-6 increment

0wl3xSec committed:
feat: Implement level-7 increment

0wl3xSec committed:
feat: Implement level-7 increment

0wl3xSec committed:
feat: Update to save tasks whenever it changes

0wl3xSec committed:
feat: Update to save tasks whenever it changes

0wl3xSec committed:
feat: Implement level-8 increment

0wl3xSec committed:
feat: Implement level-8 increment

0wl3xSec committed:
refactor: Make code more OOP

0wl3xSec committed:
Merge branch 'A-MoreOOP'

0wl3xSec committed:
refactor: Move classes to package

0wl3xSec committed:
Merge branch 'add-gradle-support'

0wl3xSec committed:
test: Add test for TaskList, Todo

0wl3xSec committed:
Merge branch 'A-JUnit'

0wl3xSec committed:
docs: Edit .gitignore

0wl3xSec committed:
docs: Add JavaDocs

0wl3xSec committed:
style: Tweak code to adhere conventions

0wl3xSec committed:
feat: Implement level-9 increment

0wl3xSec committed:
Merge branch 'branch-A-JavaDoc'

0wl3xSec committed:
Merge branch 'branch-A-CodingStandard'

0wl3xSec committed:
Merge branch 'branch-Level-9'

0wl3xSec committed:
fix: Fix exit status 1 bug

0wl3xSec committed:
fix: Fix case sensitivity of commands

0wl3xSec committed:
docs: Update JavaDocs for methods

nreHieW commented on src/main/java/pooh/Deadline.java:
```diff
@@ -0,0 +1,110 @@
+ * </p>
+ */
+public class Deadline extends Task {
+    private final String deadlineString;
```
I like how you named the variables here. They maintain the coding standard and are easy to understand at first glance what they represent.

nreHieW commented on src/main/java/pooh/Deadline.java:
```diff
@@ -0,0 +1,110 @@
+     * @return A string containing the task's details in a format suitable for file storage.
+     */
+    public String writeTaskToFile() {
+        return String.format("%s | %s | %s | %s", "D",
```
Is there any reason you chose to use String.format rather than a String concatenation here? Perhaps, it might be clearer that way? Still good use of linebreaks to make the code readable! 👍

nreHieW commented on src/main/java/pooh/Task.java:
```diff
@@ -0,0 +1,117 @@
+ */
+public abstract class Task {
+    private final String description;
+    private boolean isDone;
```
Good use of coding standard here when naming boolean variables!

nreHieW commented on src/main/java/pooh/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+     * @throws EmptyTaskDescriptorsException If the task description is empty.
+     */
+    public static void addTask(TaskList taskList, String userAction, String cmd) throws
+            EmptyTaskDescriptorsException {
```
Good to see that you abide by the line length < 120 characters standard here 👍

nreHieW commented on src/main/java/pooh/Parser.java:
```diff
@@ -0,0 +1,76 @@
+ */
+public class Parser {
+
+    /**
```
Very well written Javadoc comments!

nreHieW commented on src/main/java/pooh/Parser.java:
```diff
@@ -0,0 +1,76 @@
+            Ui.printTasksMsg(taskList);
+        } else {
+            switch (userAction.toLowerCase()) {
+            case "list":
```
Good to see that you avoided the common indentation mistake for switch statements!

nreHieW reviewed (approved):
Overall, I found your code easy to read. There were not many cases of code that had too much nesting and the coding standard was abided by diligently. Good to see that you also did not make some of the common coding standard mistakes discussed in class! LGTM 😄

0wl3xSec committed:
refactor: Add JavaFX

0wl3xSec committed:
feat: Implement level-10 increment

0wl3xSec committed:
Merge branch 'branch-Level-10'

kanna-1 commented on src/main/java/pooh/Ui.java:
```diff
@@ -0,0 +1,117 @@
+     *
+     * @return The exit message.
+     */
+    public String getExitMsg() {
```
Should this be renamed to `getExitMessage` for clarity?

kanna-1 commented on src/main/java/pooh/Ui.java:
```diff
@@ -0,0 +1,117 @@
+     * @param taskList The list of tasks.
+     * @return A formatted string representation of tasks.
+     */
+    public String getTasksMsg(TaskList taskList) {
```
Similar to the above.

kanna-1 reviewed (commented):
Excellent code.  could only find a minor coding standard violation based on the naming conventions. Overall good work! 😄

0wl3xSec committed:
feat: Add assertions to codebase

Documents important assumptions that should hold at various points in the code.

* Ensured filePath is not empty

* Ensured formatting of deadline as LocalDate object did not run into any error

0wl3xSec committed:
refactor: Improve code quality

Mainly revised the methods naming in Ui class as per PR review comments.

Methods renamed:

* getExitMsg() -> getExitMessage()

* getTasksMsg() -> getTasksMessage()

* getKeywordTasksMsg() -> getKeywordTasksMessage()

* getNoKeywordTasksFoundMsg() -> getNoKeyWordTasksFoundMessage()

* getTaskDoneMsg() -> getTaskDoneMessage()

* getTaskUndoneMsg() -> getTaskUndoneMessage()

* getAddTaskMsg() -> getAddTaskMessage()

* getDeleteTaskMsg() -> getDeleteTaskMessage()

0wl3xSec committed:
Merge pull request #2 from Zjinnnn/branch-A-Assertions

feat: Add assertions to codebase

0wl3xSec committed:
Merge branch 'master' into branch-A-CodeQuality

0wl3xSec committed:
Merge pull request #3 from Zjinnnn/branch-A-CodeQuality

refactor: Improve code quality

0wl3xSec committed:
feat: Implement B-Snooze extension

Provide a way to postpone deadlines for better convenience.

0wl3xSec committed:
Merge pull request #4 from Zjinnnn/branch-B-Snooze

feat: Implement B-Snooze extension

0wl3xSec committed:
docs: Add User Guide

0wl3xSec committed:
docs: Update User Guide

0wl3xSec committed:
docs: Add Ui.png

0wl3xSec committed:
refactor: Update UI response for find command

0wl3xSec committed:
docs: Update User Guide
